docs/datamodel/build-a-datamodel.rst

```diff
@@ -0,0 +1,627 @@
+- Landing page template
+
+The templates are stored in two different folders (``static`` and
+``templates``):
+
+.. code-block:: shell
+
+    |-- my_datamodel
+    |   |-- static
+    |   |   `-- templates
+    |   |       `-- my_datamodel
+    |   |           `-- results.html
+    |   |-- templates
+    |   |   `-- my_datamodel
+    |   |       `-- record.html
+
+
+**Search result template**
+
+The Invenio search interface is run by a JavaScript application, and thus the
+template is rendered client side in the user's browser. The template uses data
+received by the REST API and thus your REST API must be able to deliver all
+information you would like to render in the template.
+
+**Landing page template**
+
+The landing page for a single record is rendered on the server-side using a
+Jinja template.
+
+Configure the UI
+----------------
+Last step after having defined all the different schemas, serializers, loaders
+and templates is to configure your REST API and landing pages for your records.
+
+This is all done from the ``config.py``:
+
+.. code-block:: shell
+
+    |-- my_datamodel
+    |   |-- config.py
+
+**Landing page**
+
+Let's start by configuring the landing page:
+
+.. code-block:: python
+
+    RECORDS_UI_ENDPOINTS = {
+        'recid': {
+            'pid_type': 'recid',
+            'route': '/records/<pid_value>',
+            'template': 'my_datamodel/record.html',
+        },
+    }
+
+Here an explanation of the different keys:
+
+* ``pid_type``: Defines the persistent identifier type which the resolver
+  should use to lookup records. Invenio provides an internal persistent
+  identifier type called ``recid`` which is an auto-incrementing integer.
+* ``route``: URL endpoint under which to expose the landing pages.
+* ``template``: Template to use when rendering the landnge page.
+* ``recid``: Unique name of the endpoint. If this is the primary landing page,
+  it must be named the same as the value of ``pid_type`` (i.e. ``recid``).
+
+Configure the REST API
+----------------------
+Configuring the REST API is done similarly to the landing pages via the
+``RECORDS_REST_ENDPOINTS`` configuration variable in ``config.py``:
+
+**Persistent identifier type**
+
+First you provide the persistent identifier type used by the resolver. You also
+need to configure a persistent identifier minter and fetcher. In the scaffolded
+data model package, you are just using the already provided ``recid`` minter
+and fetchers.
+
+A *minter* is responsible for generating a new persistent identifier for your
+record, while a *fetcher* is responsible for extracting the persistent
+identifier from your search results:
+
+.. code-block:: python
+
+    RECORDS_REST_ENDPOINTS = {
+        'recid': dict(
+            pid_type='recid',
+            pid_minter='recid',
+            pid_fetcher='recid',
+            # ...
+        ),
+    }
+
+**Search**
+
+Next, you define the Elasticsearch index to use for searchers. The index is
```

Comment:
Next, you define the Elasticsearch index to use [for searching|to search|for the searches]?